### PR TITLE
feat(hooks): add useSuggestChainAndConnect

### PR DESCRIPTION
**Pull request by codingki**
## Description

This PR add `useSuggestChainAndConnect` hook to graz

## Checklist

- [x] I have made sure the upstream branch for this PR is correct
- [x] I have made sure this PR is ready to merge
- [x] I have made sure that I have assigned reviewers related to this project

## Changes

- [x] Added useSuggestChainAndConnect hook
- [x] update example app
- [x] update API docs

## Screenshots

ezgif-3-166d1e242b

### CODE REVIEW

File: packages/graz/src/actions/chains.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 export async function suggestChain(chainInfo: ChainInfo) {
   const keplr = getKeplr();
   await keplr.experimentalSuggestChain(chainInfo);
   return chainInfo;
 }
+
+export async function suggestChainAndConnect(chainInfo: ChainInfo) {
+  await suggestChain(chainInfo);
+  await connect(chainInfo);
+  return chainInfo;
```
Comment (grikomsn) [inline packages/graz/src/actions/chains.ts]: We might need to return both results from `connect` and `suggestChain`

File: packages/graz/src/hooks/chains.ts [hunk 2]
```diff
@@ -28,3 +28,23 @@ export function useSuggestChain({ onError, onLoading, onSuccess }: UseSuggestCha
...
+    onSuccess: (chainInfo) => Promise.resolve(onSuccess?.(chainInfo)),
+  });
+
+  return {
+    error: mutation.error,
+    isLoading: mutation.isLoading,
+    isSuccess: mutation.isSuccess,
+    suggestAndConnect: mutation.mutate,
+    suggestAndConnectAsync: mutation.mutateAsync,
+    status: mutation.status,
```
Comment (grikomsn) [inline packages/graz/src/hooks/chains.ts]: Sort props and add `isSupported` like on `useConnect`[^1]

[^1]: https://github.com/strangelove-ventures/graz/blob/1eca580fe58084369d6bfcb686c5f3d951118e6d/packages/graz/src/hooks/account.ts#L79-L85

File: example/constants/osmosis-testnet.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    bech32PrefixAccPub: "osmopub",
+    bech32PrefixValAddr: "osmovaloper",
+    bech32PrefixValPub: "osmovaloperpub",
+    bech32PrefixConsAddr: "osmovalcons",
+    bech32PrefixConsPub: "osmovalconspub",
+  },
+  currencies: [OSMO],
+  feeCurrencies: [OSMO],
+  coinType: 118,
+};
```
Comment (grikomsn) [inline example/constants/osmosis-testnet.ts]: While this serves as an example, might be best to move chain data to `src/chains.ts`[^1]

[^1]: https://github.com/strangelove-ventures/graz/blob/dev/packages/graz/src/chains/osmosis.ts

**Review by grikomsn — CHANGES_REQUESTED** [review]

File: example/constants/osmosis-testnet.ts (hunk 3 above)
Comment (codingki) [reply example/constants/osmosis-testnet.ts]: should it live in `defaultChains`?

Comment (codingki) [reply example/constants/osmosis-testnet.ts]: or we define another, like `defaultTestnetChains`?

Comment (codingki) [reply example/constants/osmosis-testnet.ts]: if we going to define it on `defaultTestnetChains` I have more testnet chains that I can put it there, structure the directory -> 
`/chains/mainet/`
`/chains/testnet/`

Comment (grikomsn) [reply example/constants/osmosis-testnet.ts]: Maybe `chains/mainnet.ts` and `chains/testnet.ts`? Then exports would be `mainnetChains` and `testnetChains`?

Comment (codingki) [reply example/constants/osmosis-testnet.ts]: ~~agree~~

I'd rather keep the chain as files, I'll push a commit and let me know what you think

File: packages/graz/src/actions/chains.ts [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
 }
+
+export async function suggestChainAndConnect(chainInfo: ChainInfo) {
+  const [resSuggest, resConnect] = await Promise.all([suggestChain(chainInfo), connect(chainInfo)]);
+
+  return {
+    chain: resSuggest,
+    account: resConnect,
+  };
+}
```
Comment (grikomsn) [inline packages/graz/src/actions/chains.ts:19]:
```suggestion
export async function suggestChainAndConnect(chainInfo: ChainInfo) {
  const [account, chain] = await Promise.all([connect(chainInfo), suggestChain(chainInfo)]);
  return { account, chain };
}
```

File: API.md [hunk 5]
```diff
@@ -411,3 +418,65 @@ function App(){
...
+const osmosisTestnet = {
+  rpc: "https://testnet-rpc.osmosis.zone",
+  rest: "https://testnet-rest.osmosis.zone",
+  chainId: "osmo-test-4",
+  chainName: "Osmosis Testnet",
+  stakeCurrency: OSMO,
+  bip44: {
+    coinType: 118,
+  },
+  bech32Config: bech32Config: Bech32Address.defaultBech32Config("osmo"),
```
Comment (grikomsn) [inline API.md]:
```suggestion
  bech32Config: Bech32Address.defaultBech32Config("osmo"),
```

**Review by grikomsn — CHANGES_REQUESTED** [review]

File: packages/graz/src/actions/chains.ts (hunk 4 above)
Comment (grikomsn) [inline packages/graz/src/actions/chains.ts]:
```suggestion
export async function suggestChainAndConnect(chainInfo: ChainInfo) {
  const chain = await suggestChain(chainInfo);
  const account = await connect(chainInfo);

  return { account, chain };
}
```

**Review by grikomsn — CHANGES_REQUESTED** [review]

**Review by grikomsn — APPROVED** [review]

**Pull request merged**